### ISSUE TITLE
ENH: wavelet scattering implementation (?)

### ISSUE THREAD

**Issue by tylerjereddy** [scikit-image/scikit-image#8254]
I discussed this a little with @stefanv at the SciPy conference today, but I probably didn't communicate it very clearly after a long day of the conference. To provide a brief summary, there doesn't appear to be a super well maintained scientific Python ecosystem project that offers wavelet scattering on image data--the formal reference is probably https://doi.org/10.1109/TPAMI.2012.230 -- Joan Bruna and Stephane Mallat (2013) Invariant Scattering Convolution Networks. _IEEE TRANSACTIONS ON PATTERN ANALYSIS AND MACHINE INTELLIGENCE_. While working on a research project that focuses on gradient free neural networks (specifically this new class of estimator proposed for `scikit-learn`: https://github.com/scikit-learn/scikit-learn/pull/33215), we found that wavelet scattering was a top performing feature, often exceeding test set accuracy of standard `skimage` features like `skimage.feature.hog`.
 
The authors of the mansucript seem to agree that these features are quite helpful for downstream supervised machine learning, noting in their manuscript:

> State-of-the-art classification results are obtained for handwritten digits and texture discrimination, with a Gaussian kernel SVM and a generative PCA classifier

and the broader community also seems to agree, with their manuscript now at 2331 citations (Google Scholar). To be fair, there is a rather less well maintained Python library (https://github.com/kymatio/kymatio) that we are using for this at the moment (and I believe there is author overlap with the original primary paper), and the paper associated with that Python library is well cited (see BibTeX citation below the fold).

<details>

```latex
@article{JMLR:v21:19-047,
  author  = {Mathieu Andreux and TomÃ¡s Angles and Georgios Exarchakis and Roberto Leonarduzzi and Gaspar Rochette and Louis Thiry and John Zarka and StÃ©phane Mallat and Joakim AndÃ©n and Eugene Belilovsky and Joan Bruna and Vincent Lostanlen and Muawiz Chaudhary and Matthew J. Hirn and Edouard Oyallon and Sixin Zhang and Carmine Cella and Michael Eickenberg},
  title   = {Kymatio: Scattering Transforms in Python},
  journal = {Journal of Machine Learning Research},
  year    = {2020},
  volume  = {21},
  number  = {60},
  pages   = {1-6},
  url     = {http://jmlr.org/papers/v21/19-047.html}
}
```

</details>

There may be some reasons to include this in `skimage` but also some reasons not to--the `kymatio` team is obviously very knowledgeable on this topic, and the convolutional networks theme is arguably quite specialized and best handled by the experts who wrote the paper. The CNN, even if fixed rather than learned, may bring up some scoping questions for `skimage` too perhaps. Even though the convolutional networks for processing the image data used are fixed rather than learned, there are apparently compelling reasons to support autodiff workflows with i.e., `torch` ecosystem (we mostly use `kymatio.numpy` for [`Scattering2D`](https://www.kymat.io/codereference.html?highlight=scattering2d#kymatio.numpy.Scattering2D) but the existence of other package namespaces does imply some complexity). The `kymatio` package doesn't seem to be in terrible shape really, but they may have limited bandwidth (i.e., https://github.com/kymatio/kymatio/issues/1068).

Seems like a potentially tricky scope question, but Stefan did suggest at least opening an issue. I wonder if the potential value of `torch` integration becomes less scary with long term likelihood of array API support coming in the broader ecosystem anyway. [kymatio](https://github.com/kymatio/kymatio) has a BSD-3 license, so I suppose pinging them/communicating with them is also just fine if desired.  A student on our team (@eiviani-lanl) also checked and couldn't find evidence that https://github.com/PyWavelets/pywt offered support for this type of wavelet transform either. Maybe I'll just leave this here for now.
